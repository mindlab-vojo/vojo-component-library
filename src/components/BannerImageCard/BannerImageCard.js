import React from 'react'
import PropTypes from 'prop-types'

import Typography from '../Typography'

import ThemeConsumer from '../../style/ThemeManager/ThemeProvider'
import * as BannerImageCard from './BannerImageCard.style'

const BannerImageCardComponent = ({
  imageUrl,
  onClickCard,
  subtitle,
  title,
}) => {

  const renderComponent = (theme) => (
    <BannerImageCard.Wrapper
      onClick={onClickCard}
      theme={theme}>
      <BannerImageCard.Cropper>
        <BannerImageCard.Image
          src={imageUrl}/>
      </BannerImageCard.Cropper>
      <BannerImageCard.Overlay/>
      <BannerImageCard.TextArea>
        <BannerImageCard.Title>
          <Typography 
            fontSize="18px"
            fontWeight="bolder"
            color={theme.colors.whiteColor}
            fontFamily="'Montserrat', sans-serif">
            {title}
          </Typography>
        </BannerImageCard.Title>
        <BannerImageCard.Subtitle>
          <Typography 
            fontSize="12px"
            color={theme.colors.whiteColor}>
            {subtitle}
          </Typography>
        </BannerImageCard.Subtitle>
      </BannerImageCard.TextArea>
    </BannerImageCard.Wrapper>
  )
  return (
    <ThemeConsumer>
      {({ theme }) => renderComponent(theme)}
    </ThemeConsumer>
  )
}

BannerImageCardComponent.propTypes = {
  imageUrl: PropTypes.string,
  onClickCard: PropTypes.func,
  subtitle: PropTypes.string,
  title: PropTypes.string,
}

export default BannerImageCardComponent
