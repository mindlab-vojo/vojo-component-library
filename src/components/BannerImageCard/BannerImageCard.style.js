import styled from 'styled-components'

export const Wrapper = styled.div`
  border-radius: 8px;
  box-sizing: border-box;
  cursor: pointer;
  display: flex;
  flex-direction: column;
  height: 60px;
  overflow: hidden;
  position: relative;
  width: 100%;
`

export const Cropper = styled.div`
  align-items: center;
  box-sizing: border-box;
  display: flex;
  height: 100%;
  justify-content: center;
  overflow: hidden;
  width: 100%;
`

export const Image = styled.img`
  display: inline;
  margin: 0 auto;
  height: auto;
  width: 100%;
`

export const Overlay = styled.div`
  background: #391DDD;
  box-sizing: border-box;
  height: 100%;
  position: absolute;
  opacity: 0.4;
  top: 0;
  transform: matrix(-1, 0, 0, 1, 0, 0);
  width: 100%;
`
export const TextArea = styled.div`
  bottom: 0px;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  padding: 0 12px;
  position: absolute;
  top: 0px;
  width: 100%;
`

export const Title = styled.div``

export const Subtitle = styled.div``