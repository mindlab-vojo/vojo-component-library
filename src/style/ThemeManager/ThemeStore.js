import { vojo } from './../themes'

class ThemeStore {

  constructor(theme) {
    this.defaultTheme = vojo
    this.theme = theme ? theme : this.defaultTheme
  }

  configure = ({ theme }) =>  {
    if (theme) this.configureTheme(theme)
    else this.configureTheme(this.defaultTheme)
  }

  configureTheme = (theme) => {
    this.theme = theme
  }

  modify = (modifiedProps) => {
    const currentTheme = this.theme
    return { currentTheme, modifiedProps }
  }

  getTheme = () => this.theme

}

export default ThemeStore