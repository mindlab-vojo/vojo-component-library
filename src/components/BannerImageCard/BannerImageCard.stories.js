import React from 'react';
import { withKnobs, text, } from '@storybook/addon-knobs';
import { withA11y } from '@storybook/addon-a11y';

import BannerImageCard from '.'; 
import Container from '../Container';
import { ThemeProvider } from "../../../src";
import categories from '../../../.storybook/categories';

export const bannerImageCard = () => {
  const imageUrl = text('imageUrl', 'https://russelservicos.com.br/wp-content/uploads/2016/01/balconista1.jpg')
  const subtitle = text('subtitle', 'Ver detalhes da função')
  const title = text('title', 'Balconista')

  return (
    <BannerImageCard
      imageUrl={imageUrl}
      subtitle={subtitle}
      title={title}/>
  )
}

export default {
  title: categories.Images,
  component: bannerImageCard,
  decorators: [
    withKnobs, 
    withA11y, 
    (story) => 
      <ThemeProvider>
        <Container maxWidth="full">
          <Container maxWidth="desktop">
            { story() }
          </Container>
        </Container>
      </ThemeProvider>,
  ],
}

bannerImageCard.story = {
  name: 'BannerImageCard',
};
  