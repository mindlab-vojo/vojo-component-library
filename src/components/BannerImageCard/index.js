import BannerImageCard from './BannerImageCard'
export default BannerImageCard