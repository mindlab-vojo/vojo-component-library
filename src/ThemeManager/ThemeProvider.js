import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  render() {
    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer