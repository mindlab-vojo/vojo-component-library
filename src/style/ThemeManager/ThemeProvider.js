import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  getThemeFont = () => {
    const { themeConfig } = this.state
    return themeConfig.font
  }

  render() {

    const GlobalFont = this.getThemeFont()

    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        <GlobalFont/>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer