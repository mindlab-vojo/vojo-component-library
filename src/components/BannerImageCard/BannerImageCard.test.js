import React from 'react';
import { render } from '@testing-library/react';

import BannerImageCard from './';
import { ThemeProvider } from './../../style/ThemeManager/ThemeProvider';

test('Test BannerImageCard component render', () => {
  const fakeFn = jest.fn()
  const component = render(
    <ThemeProvider>
      <BannerImageCard
        imageUrl="https://russelservicos.com.br/wp-content/uploads/2016/01/balconista1.jpg"
        onClickCard={fakeFn}
        subtitle="Testando o subtÃ­tulo."
        title="Balconista"
      />
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot()
});
