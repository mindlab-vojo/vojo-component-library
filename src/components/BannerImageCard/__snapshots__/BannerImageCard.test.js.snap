// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BannerImageCard component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ eViaoB"
      >
        <div
          class="sc-gsTCUz RDkkP"
        >
          <img
            class="sc-dlfnbm cfaUiR"
            src="https://russelservicos.com.br/wp-content/uploads/2016/01/balconista1.jpg"
          />
        </div>
        <div
          class="sc-hKgILt dxKtqN"
        />
        <div
          class="sc-eCssSg boVEaw"
        >
          <div
            class="sc-jSgupP"
          >
            <p
              class="sc-iBPRYJ eTMuIg"
            >
              Balconista
            </p>
          </div>
          <div
            class="sc-gKsewC"
          >
            <p
              class="sc-fubCfw cJpRWy"
            >
              Testando o subtítulo.
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ eViaoB"
    >
      <div
        class="sc-gsTCUz RDkkP"
      >
        <img
          class="sc-dlfnbm cfaUiR"
          src="https://russelservicos.com.br/wp-content/uploads/2016/01/balconista1.jpg"
        />
      </div>
      <div
        class="sc-hKgILt dxKtqN"
      />
      <div
        class="sc-eCssSg boVEaw"
      >
        <div
          class="sc-jSgupP"
        >
          <p
            class="sc-iBPRYJ eTMuIg"
          >
            Balconista
          </p>
        </div>
        <div
          class="sc-gKsewC"
        >
          <p
            class="sc-fubCfw cJpRWy"
          >
            Testando o subtítulo.
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
